src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -218,7 +217,7 @@ public static bool IsValidEnumType(this TypeSymbol type)
         {
             var underlyingType = type.GetEnumUnderlyingType();
             // SpecialType will be None if the underlying type is invalid.
-            return (underlyingType is object) && (underlyingType.SpecialType != SpecialType.None);
```

Comment:
>underlyingType.SpecialType != SpecialType.None [](http://example.com/codeflow?start=50&length=46)

This logic also looks questionable, in fact we have a `SpecialType` extension called ```IsValidEnumUnderlyingType```, which we should probably use here. I will open a follow up issue for that.

Reply:
Opened #72317